website_seo_redirection/models/website_seo_redirection.py

```diff
@@ -105,7 +105,9 @@ def find_origin(self, redirected_path=None):
             ("destination", "=", path),
             ("relocate_controller", "=", True),
         ])
-        if not redirection.origin:
+        if len(redirection) > 1:
+            raise NoOriginError(_("Many origin found for this redirection."))
```

Comment:
I understand the issue, but I think you are not addressing it right.

Here you are replacing one exception by another, but any exception will get the user to a 500 error anyway.

I think the correct path is to add a constraint that does not allow the web admin to create >1 redirections to the same destination if any of them has `relocate_controller==True`. This way, you will never get to this situation and then no exception will be raised.